### ISSUE TITLE
[Enhancement]: Add support for AWS AppSync merged APIs

### ISSUE THREAD

**Issue by ryancausey** [hashicorp/terraform-provider-aws#33148]
### Description

Add support for AWS AppSync merged APIs.

### Affected Resource(s) and/or Data Source(s)

* aws_appsync_graphql_api
* aws_appsync_api_association # new resource type.

### Potential Terraform Configuration

```terraform
resource "aws_appsync_graphql_api" "example" {
  authentication_type = "API_KEY"
  name                = "example"
  api_type            = "GRAPHQL"
}

resource "aws_appsync_graphql_api" "test_merged_api" {
  authentication_type = "API_KEY"
  name                = "test_merged_api"
  api_type            = "MERGED"
  merged_api_execution_role_arn = aws_iam_role.test_merged_api_role.arn
}

# new resource type.
resource "aws_appsync_api_association" "test_api_association" {
  source_api_id = aws_appsync_graphql_api.example.id
  merged_api_id = aws_appsync_graphql_api.test_merged_api.id
  merge_type    = "MANUAL_MERGE | AUTO_MERGE"
}
```

### References

https://docs.aws.amazon.com/en_us/appsync/latest/devguide/merged-api.html
https://docs.aws.amazon.com/appsync/latest/APIReference/API_CreateGraphqlApi.html
https://docs.aws.amazon.com/appsync/latest/APIReference/API_AssociateMergedGraphqlApi.html
https://docs.aws.amazon.com/appsync/latest/APIReference/API_AssociateSourceGraphqlApi.html

### Would you like to implement a fix?

None

**Comment by ryancausey** [comment]
I was going to see if I could use the awscc Terraform provider for this, but I found out that AWS CloudControl API does not support these resources either. So, there's currently no Terraform based resources for this in either of the providers.

https://docs.aws.amazon.com/cloudcontrolapi/latest/userguide/supported-resources.html

**Comment by hans-d** [comment]
https://docs.aws.amazon.com/appsync/latest/APIReference/API_CreateGraphqlApi.html#API_CreateGraphqlApi_RequestSyntax
> [apiType](https://docs.aws.amazon.com/appsync/latest/APIReference/API_CreateGraphqlApi.html#API_CreateGraphqlApi_RequestSyntax)
The value that indicates whether the GraphQL API is a standard API (GRAPHQL) or merged API (MERGED).
Type: String
Valid Values: GRAPHQL | MERGED
Required: No

**Comment by drewtul** [comment]
The merged api type was #39159 I'll be following up with the new resource to create the associations.
